File: cpp/program/Supermarket.cpp
#include <iostream>
#include "Perusahaan.h"
using namespace std;

/* Class Supermarket yang mewarisi Perusahaan */
class Supermarket : public Perusahaan{
    protected:
        string kodeSupermarket;   // Kode unik supermarket
        string lokasi;            // Lokasi supermarket
        string luasToko;          // Luas toko (misalnya "2000 m2")
        string jamOperasional;    // Jam operasional (misal: "08:00 - 22:00")

    public:
        // Constructor kosong
        Supermarket() : Perusahaan() {
            kodeSupermarket = "";
            lokasi = "";
            luasToko = "";
            jamOperasional = "";
        }

        // Constructor lengkap
        Supermarket(string noHp, string email, string alamat,
                    string nama, int tahunBerdiri, string bidangUsaha,
                    string kodeSupermarket, string lokasi, string luasToko, string jamOperasional)
                    : Perusahaan(noHp, email, alamat, nama, tahunBerdiri, bidangUsaha) {
            this->kodeSupermarket = kodeSupermarket;
            this->lokasi = lokasi;
            this->luasToko = luasToko;
            this->jamOperasional = jamOperasional;
        }

        // Getter dan Setter
        string getKodeSupermarket() {
            return kodeSupermarket;
        }

        void setKodeSupermarket(string kodeSupermarket) {
            this->kodeSupermarket = kodeSupermarket;
        }

        string getLokasi() {
            return lokasi;
        }

        void setLokasi(string lokasi) {
            this->lokasi = lokasi;
        }

        string getLuasToko() {
            return luasToko;
        }

        void setLuasToko(string luasToko) {
            this->luasToko = luasToko;
        }

        string getJamOperasional() {
            return jamOperasional;
        }

        void setJamOperasional(string jamOperasional) {
            this->jamOperasional = jamOperasional;
        }
        ~Supermarket(){}
};

File: python/program/Supermarket.py
from Perusahaan import Perusahaan

# subclass Supermarket
class Supermarket(Perusahaan):

    def __init__(self, noHp: str, email: str, alamat: str,
                 nama: str, tahunBerdiri: int, bidangUsaha: str,
                 kodeSupermarket: str, lokasi: str, luasToko: str, jamOperasional: str):
        # panggil constructor parent (Perusahaan)
        super().__init__(noHp, email, alamat, nama, tahunBerdiri, bidangUsaha)
        # atribut khusus Supermarket
        self._kodeSupermarket = str(kodeSupermarket)
        self._lokasi = str(lokasi)
        self._luasToko = str(luasToko)
        self._jamOperasional = str(jamOperasional)

    # Getter & Setter Kode Supermarket
    def getKodeSupermarket(self) -> str:
        return self._kodeSupermarket

    def setKodeSupermarket(self, kodeSupermarket: str) -> None:
        self._kodeSupermarket = str(kodeSupermarket)

    # Getter & Setter Lokasi
    def getLokasi(self) -> str:
        return self._lokasi

    def setLokasi(self, lokasi: str) -> None:
        self._lokasi = str(lokasi)

    # Getter & Setter Luas Toko
    def getLuasToko(self) -> str:
        return self._luasToko

    def setLuasToko(self, luasToko: str) -> None:
        self._luasToko = str(luasToko)

    # Getter & Setter Jam Operasional
    def getJamOperasional(self) -> str:
        return self._jamOperasional

    def setJamOperasional(self, jamOperasional: str) -> None:
        self._jamOperasional = str(jamOperasional)
